sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -146,15 +146,6 @@ namespace Azure { namespace Core {
      */
     Context() : m_contextSharedState(std::make_shared<ContextSharedState>()) {}
 
-    /**
```

Comment:
I'd rather add "copy constructor = default" delcaration. More explanations: https://github.com/Azure/azure-sdk-for-cpp/pull/3957#pullrequestreview-1116185634

Reply:
I'm following the recommendation from @CaseyCarter in that PR. Could you explain why it's better to have an explicitly declared copy ctor and operator= rather than relying on the compiler to implicitly generate it (the codegen is identical, so what does the explicit declaration bring)?

Reply:
To be clear: I don't have a preference either way here. I'm honestly looking for guidance. Casey suggested removing both and letting the compilers defaults kick in (on the theory that "0 constructor declarations are better than 3 constructor definitions"). You're suggesting we make them explicit and defaulted. I'd like to understand your reasoning about why your proposal is better.

Reply:
I'm for explicitly declaring default operations with `= default` or `= delete` even when this is the default behavior. Especially in cases when it is not immediately clear, like we had in comments there, asking whether it would be a breaking change or not (you can't know for sure until you read class definition, including its members - i.e. if a class has `std::unique_ptr` it is non-copyable). So, between the approach to remove the .`..operator=(...) = default` declaration and the one to add `Context(Context const&) = default` declaration, I pick the explicit one. You read it, and it is immediately clear, no more questions.
I'd have more of `= default` and `= delete` - they can't hurt, they only make things clearer.

Reply:
I'm not saying we should always define all the operations, including clearly plain and simple classes and structs with simple types - that's an overkill that would add too much visual clutter. But when in doubt, and the class definition is more than one screen anyways - I think we should.

Reply:
I like it (as I said, I have no preferences). I'll circle with Casey about this to make sure he's in alignment.

Reply:
If possible, when all else is equal, I vote for avoiding being overly verbose on class type definitions and reduce having to write more boilerplate code. Having to write/read/review less code is a maintainability win, especially when there is little to no customer benefit in doing more work :)